packages/appium/lib/cli/extension-command.js

```diff
@@ -245,30 +276,68 @@ class ExtensionCommand {
             const msg =
               `Could not resolve ${this.type}; are you sure it's in the list ` +
               `of supported ${this.type}s? ${JSON.stringify(knownNames)}`;
-            throw new Error(msg);
+            throw this._fatalError(msg);
           }
+          probablyExtName = name;
           pkgName = this.knownExtensions[name];
           // given that we'll use the install type in the driver json, store it as
           // 'npm' now
           installType = INSTALL_TYPE_NPM;
         }
       }
+      installOpts = {installSpec, pkgName, pkgVer};
+    }
 
-      extData = await this.installViaNpm({installSpec, pkgName, pkgVer});
+    // fail fast here if we can
+    if (probablyExtName && this.config.isInstalled(probablyExtName)) {
+      throw this._fatalError(
```

Comment:
not sure if this should be an error or just a warning

Reply:
if the thing is already installed?  it was an error before.  now it's just an error...earlier.

Reply:
I could go for "if already installed, exit with a nice message instead of a scary error"